### PR TITLE
Add deprecation warnings for week-date specifiers

### PR DESCRIPTION
**Pull request by thecoop**
Week dates are changing on CLDR, so warn about that too.

Follows on from https://github.com/elastic/elasticsearch/pull/112917

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine rerun elasticsearch-ci/packaging-tests-windows-sample

**Review by rjernst — APPROVED** [review]
Looks good, one question

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -77,6 +77,9 @@ tasks.named("yamlRestTestV7CompatTest").configure {
     'ml/evaluate_data_frame/Test outlier_detection auc_roc include curve',
     'ml/evaluate_data_frame/Test classification auc_roc',
     'ml/evaluate_data_frame/Test classification auc_roc with default top_classes_field',
+    'security/authz/13_index_datemath/Test indexing documents with datemath, when forbidden',
```
[the remark this answers is not in the document]
Comment (thecoop) [reply x-pack/plugin/build.gradle:80]: The warnings are causing failures with previous yaml test definitions

**Comment by thecoop** [comment]
Windows packaging tests are https://github.com/elastic/elasticsearch/issues/113177

**Pull request merged**